src/main/pages/che-7/extensions/assembly_telemetry.adoc

```diff
@@ -0,0 +1,297 @@
+---
+title: Telemetry
+keywords: [telemetry]
+tags: [telemetry]
+sidebar: che_7_docs
+permalink: che-7/telemetry/
+folder: che-7/extensions
+summary:
+---
+
+:page-liquid:
+:parent-context-of-telemetry: {context}
+
+[id="eclipse-che4z_{context}"]
+= Telemetry
+
+Telemetry is the transparent and ethical collection of the usage data. By default, telemetry is not available in the {prod}, 
+but there is an abstract API that allows enabling telemetry via the plugin mechanism. This approach is used in the link:{site-baseurl}che-7/hosted-che/[Hosted Che] service where telemetry is enabled for every workspace.
+
+== Use cases
+
+Eclipse Che telemetry API allows tracking:
+
+* Duration of workspace usage.
+* User-driven actions like file editing, committing and pushing to the remote repositories
+* The list of plugins enabled in the workspace
+* Programming languages and link:{site-baseurl}che-7/making-a-workspace-portable-using-a-devfile/devfiles[devfiles] used in the workspaces
+
+== How it works?
+
+When a {prod-short} workspace starts, the che-theia container starts the telemetry plugin, which is responsible for sending telemetry events to a backend.  If the `$CHE_WORKSPACE_TELEMETRY_BACKEND_PORT` environment variable is set in the workspace pod, the telemetry plugin will send events to a backend listening at that port.
+
+If the {prod-short} workspace has a telemetry backend container running, and it is listening on `$CHE_WORKSPACE_TELEMETRY_BACKEND_PORT` it will take those events sent from the telemetry plugin, turn them into the backend-specific representation of events, and send them to the configured analytics backend (e.g. Segment, Woopra, etc.).
+
+image::telemetry/telemetry_diagram.png[]
+
+We will create an `AnalyticsManager` class that extends `AbstractAnalyticsManager` and implements the following methods:
```

Comment:
would be nice to link it via url https://github.com/che-incubator/che-workspace-telemetry-client/blob/master/backend-base/src/main/java/org/eclipse/che/incubator/workspace/telemetry/base/AbstractAnalyticsManager.java

Reply:
Fixed